docs/adv-topics/autopatch.md

```diff
@@ -66,64 +161,279 @@ context = autopatch					; dialing context for phone
 that users can bring up or take down the autopatch:
 ```
 [functions]
-61 = autopatchup,noct=1,farenddisconnect=1,dialtime=20000,context=autopatch
-62 = autopatchdn 
+61 = autopatchup,noct=1,farenddisconnect=1,dialtime=20000,context=autopatch,quiet=1
+0 = cmd,/var/lib/asterisk/hangupPhones            ; Hangup All phone patches
+```
+
+#### Autopatch Options
+
+There are several options that can be passed to the autopatchup command class.
+
+This table summarizes what they do:
+
+| Autopatch Option | Description |
+|---------|-------------|
+| context | Override the context specified for the autopatch in rpt.conf |
+| dialtime | The maximum time to wait between DTMF digits when a telephone number is being dialed. The patch will automatically disconnect if this time is exceeded. The value is specified in milliseconds. |
+| farenddisconnect | When set to 1, the patch will automatically disconnect when the called party hangs up. The default is send a circuit busy tone until the radio user brings the patch down. |
+| noct | When this is set to 1, the courtesy tone during an autopatch call will be disabled. The default is to send the courtesy tone whenever the radio user unkeys. |
+| quiet | When set to 1, do not send dial tone, voice responses, just try to connect the call. |
+| voxalways | When set to 1, enables vox mode. |
+| exten | Overrides the default autopatch extension. |
+| nostar | Disables the repeater function prefix |
+
+#### Create a Script to Handle Terminating Calls
+Create the following script to handle terminating calls. Without this script, you will be unable to forcefully terminate **incoming** calls:
+
+```bash
+nano /var/lib/asterisk/hangupPhones
 ```
 
-In **extensions.conf**, a stanza for the autopatch context is added which 
-refers to the peer stanza in iax.conf or sip.conf. 
+Paste this script into the file:
 
-For outbound IAX connections extensions.conf
+```bash
+#!/bin/bash
+# Target pattern to match (adjust as needed for different calls)
+TARGET_PATTERN="IAX2/voipms-"
+# TARGET_PATTERN="PJSIP/flowroute-"
+# Get the list of channels and their concise details
+CHANNELS=$(asterisk -rx "core show channels concise")
+echo "Checking for matching channels..."
+echo "--------------------------------"
+# Iterate over each line to find the target channel
+while IFS= read -r line; do
+    if [[ $line == $TARGET_PATTERN* ]]; then
+        CHANNEL=$(echo "$line" | awk -F'!' '{print $1}')
+        CALLERID=$(echo "$line" | awk -F'!' '{print $3}')
+        echo "Found matching channel: $CHANNEL with Caller ID: $CALLERID"
+        echo "Hanging up channel: $CHANNEL"
+        asterisk -rx "channel request hangup $CHANNEL"
+    fi
+done <<< "$CHANNELS"
 ```
-[autopatch] 
 
-exten => _1NXXNXXXXXX,1,Dial,IAX2/peername/${EXTEN} 
-exten => _1NXXNXXXXXX,2,Congestion 
+Make it executable:
+
+```bash
+chmod +x /var/lib/asterisk/hangupPhones
 ```
-For outbound SIP connections it should look like this:
+
+### extentions.conf
```

Comment:
extentions.conf --> extensions.conf

... and do you want to use the full path?